Android/app/src/androidTest/java/tie/hackathon/travelguide/Tests/DemoTest.java

```diff
@@ -1,19 +1,34 @@
 package tie.hackathon.travelguide.Tests;
 
+import org.junit.Assert;
 import org.junit.Test;
 
-import static android.support.test.espresso.Espresso.onView;
-import static android.support.test.espresso.action.ViewActions.click;
-import static android.support.test.espresso.matcher.ViewMatchers.withContentDescription;
+import tie.hackathon.travelguide.Constants.Strings;
+import tie.hackathon.travelguide.Constants.Timeouts;
+import tie.hackathon.travelguide.R;
+import tie.hackathon.travelguide.Screens.LoginPage;
+import tie.hackathon.travelguide.Screens.MainPage;
 
 /**
  * Created by valentin.boca on 2/21/2018.
  */
 
 public class DemoTest extends EspressoTestBase {
 
-    @Test
-    public void testDemo() throws Exception {
-        onView(withContentDescription("SIGNUP")).perform(click());
+    @Test(timeout = Timeouts.TEST_TIMEOUT_MEDIUM)
+    public void testIfSignUpIsSuccessful() throws Exception {
+        LoginPage.signUp(Strings.USER, Strings.PHONE_NO, Strings.PASSWORD);
+        LoginPage.tapSignUpButton();
+        Assert.assertTrue(Strings.MAIN_PAGE_NOT_DISPLAYED, MainPage.isScreenDisplayed(R.id.cityname));
```

Comment:
"isScreenDisplayed", la fel, nu-mi zice nimic. 
"La ce screen te referi?" -> trebuie sa regasesc raspunsul in numele metodei. :)